File: app/src/main/java/bzh/naoned/naonedbusdatadisplay/viewmodel/TrafficViewModel.java
package bzh.naoned.naonedbusdatadisplay.viewmodel;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.databinding.BindingAdapter;
import android.support.annotation.NonNull;
import android.widget.ImageView;

import com.squareup.picasso.Picasso;

import bzh.naoned.naonedbusdatadisplay.service.model.TrafficData;
import bzh.naoned.naonedbusdatadisplay.service.repository.TrafficRepository;

public class TrafficViewModel extends AndroidViewModel {
    private TrafficData trafficData;

    public TrafficViewModel(@NonNull Application application) {
        super(application);
    }

    @BindingAdapter({"imageUrl"})
    public static void loadImage(ImageView view, String imageUrl) {
        Picasso.get()
                .load(imageUrl)
                .into(view);
    }

    public TrafficData getTraffic(String id) {
        if (trafficData == null)
            trafficData = TrafficRepository.getInstance().getTraffic(id);

        return trafficData;
    }
}

File: app/src/main/java/bzh/naoned/naonedbusdatadisplay/viewmodel/TrafficViewModel.kt
package bzh.naoned.naonedbusdatadisplay.viewmodel

import android.app.Application
import android.arch.lifecycle.AndroidViewModel
import android.databinding.BindingAdapter
import android.widget.ImageView

import com.squareup.picasso.Picasso

import bzh.naoned.naonedbusdatadisplay.service.model.TrafficData
import bzh.naoned.naonedbusdatadisplay.service.repository.TrafficRepository

class TrafficViewModel(application: Application) : AndroidViewModel(application) {
    private var trafficData: TrafficData? = null
    private val trafficRepo : TrafficRepository = TrafficRepository.getInstance()

    fun getTraffic(id: String): TrafficData? {
        if (trafficData == null)
            trafficData = trafficRepo.getTraffic(id)

        return trafficData
    }

    @BindingAdapter("imageUrl")
    fun loadImage(view: ImageView, imageUrl: String) {
        Picasso.get()
                .load(imageUrl)
                .into(view)
    }
}
